### PR TITLE
XW-887 | Work towards ESLint

### PR DESCRIPTION
**Pull request by hakubo**
- a bit of cleanup in tests
- ember-cli-eslint introduced
 - I can't really test that before we have working tests
 - but I don't have time now to work more on that
 - I'll have bit more time this Friday
 - feel free to take that PR!
- ember-cli-qunit-eslint might be better but again can't really tell
 - problem I have found with this is that when ember server is used on file change jshint is used (sic!)
- no conflict is not loaded in tests
 - so we could use there require/define 

(Sorry picked wrong ticket for branch name :()

### CODE REVIEW

File: front/main/ember-cli-build.js [hunk 1]
```diff
@@ -53,7 +53,11 @@ module.exports = function (defaults) {
 	app.import(app.bowerDirectory + '/visit-source/dist/visit-source.js');
 	app.import(app.bowerDirectory + '/Autolinker.js/dist/Autolinker.min.js');
 	app.import(app.bowerDirectory + '/ember-performance-sender/dist/ember-performance-sender.js');
-	app.import('vendor/loader-no-conflict.js');
+
+	if (app.env !== 'test') {
+		app.import('vendor/loader-no-conflict.js');
```
Comment (rogatty) [inline front/main/ember-cli-build.js]: After https://github.com/Wikia/mercury/commit/2fc2f9bae1a54c72e25f841ed828c6be2f6ecb21 it's no longer used. We should set `app.options.derequire.enabled = false` but I'm not sure what's the API for doing that.

File: front/main/package.json [hunk 2]
```diff
@@ -26,18 +26,18 @@
     "broccoli-concat": "^2.0.4",
     "broccoli-funnel": "^1.0.1",
     "broccoli-merge-trees": "^1.1.0",
-    "ember-cli": "^2.2.0-beta.1",
+    "ember-cli": "^2.2.0-beta.2",
```
Comment (rogatty) [inline front/main/package.json]: According to https://github.com/ember-cli/ember-cli/blob/master/CHANGELOG.md#220-beta2 we should also do this: http://ember-cli.com/user-guide/#project-update.

**Comment by tomasznapieralski** [comment]
few tests are passing now, but there is weird situation sometimes when all tests passed 20/20 but error occurs `Callback was already called` i found that this is something which many people have:

https://github.com/caolan/async/search?q=Callback+was+already+called&type=Issues&utf8=✓

for now i don't know what we should do with this

**Comment by hakubo** [comment]
it was a bug in testem
https://github.com/ember-cli/ember-cli/issues/5288
fixed already

File: front/main/app/index.html [hunk 3]
```diff
@@ -1,15 +1,15 @@
 <!DOCTYPE html>
 <html lang="{{wikiVariables.language.content}}" dir="{{wikiVariables.language.contentDir}}">
-<head>
-	<meta charset="utf-8">
-	<meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <head>
```
Comment (rogatty) [inline front/main/app/index.html]: Use tabs

File: front/main/app/initializers/ember-hammer.js [hunk 4]
```diff
@@ -1,3 +1,8 @@
+/**
+ * Initializer for Ember Hammer
```
Comment (rogatty) [inline front/main/app/initializers/ember-hammer.js]: `initializers/ember-hammer.js`
Thanks, cpt. Obvious!

File: front/main/app/templates/application.hbs [hunk 5]
```diff
@@ -14,7 +14,7 @@
 	noScroll=noScroll
 	sideNavVisible=sideNavVisible
 	smartBannerVisible=smartBannerVisible
-	themeBar=themeBar
+	themeBar=themeBarem
```
Comment (rogatty) [inline front/main/app/templates/application.hbs]: :no_good:

File: front/main/bower.json [hunk 6]
```diff
@@ -28,7 +26,8 @@
     "visit-source": "Wikia/visit-source#0.1.1",
     "weppy": "Wikia/weppy#0.9.3",
     "wikia-style-guide": "Wikia/style-guide#v1.8.2",
-    "Autolinker.js": "~0.22.0"
+    "Autolinker.js": "0.22.0",
+    "sinonjs": "~1.17.1"
```
Comment (rogatty) [inline front/main/bower.json]: Keep the alphabetic order.

Comment (tomasznapieralski) [inline front/main/bower.json]: `npm install --save-dev` did that, i think we should keep as he wants

Comment (rogatty) [inline front/main/bower.json]: Maybe it's because **A**utolinker?

Comment (tomasznapieralski) [inline front/main/bower.json]: ¯\_(ツ)_/¯

**Comment by tomasznapieralski** [comment]
All tests are passing, app is building

**Pull request merged**